### ISSUE TITLE
How'sMySSL rating: bad

### ISSUE THREAD

**Issue by idotta** [cpp-netlib/cpp-netlib#689]
Using https://www.howsmyssl.com/a/check api returns a bad rating to a simple TLS client. Is there something I can tweak to improve it?

**Comment by anonimal** [comment]
@idotta what ciphers are being reported and what options are you currently using for your client?

**Comment by idotta** [comment]
hey @anonimal I'm using the hello_world_client, no modifications, just to test the library. Here's part of the response:

"insecure_cipher_suites": {
"TLS_DHE_DSS_WITH_DES_CBC_SHA":["uses keys smaller than 128 bits in its encryption"],
"TLS_DHE_RSA_WITH_DES_CBC_SHA":["uses keys smaller than 128 bits in its encryption"],
"TLS_ECDHE_ECDSA_WITH_RC4_128_SHA":["uses RC4 which has insecure biases in its output"],
"TLS_ECDHE_RSA_WITH_RC4_128_SHA":["uses RC4 which has insecure biases in its output"],
"TLS_ECDH_ECDSA_WITH_RC4_128_SHA":["uses RC4 which has insecure biases in its output"],
"TLS_ECDH_RSA_WITH_RC4_128_SHA":["uses RC4 which has insecure biases in its output"],
"TLS_RSA_WITH_DES_CBC_SHA":["uses keys smaller than 128 bits in its encryption"],
"TLS_RSA_WITH_RC4_128_MD5":["uses RC4 which has insecure biases in its output"],
"TLS_RSA_WITH_RC4_128_SHA":["uses RC4 which has insecure biases in its output"]
}

**Comment by anonimal** [comment]
@idotta The bad news: there is no adequate documentation on how to select ciphers nor is there a const type array or container of ciphers to choose from (someone, please correct me if that's wrong). When this feature was merged, I think the assumption was made that users would implement according to [OpenSSL's underlying syntax](https://www.openssl.org/docs/manmaster/apps/ciphers.html).

The good news: https://github.com/cpp-netlib/cpp-netlib/pull/530 gives a good run-down of how to use ciphers with client options.

TL;DR: you can try, at a minimum (and depending on use-case) something like this:

```cpp
boost::network::http::client::options client_options;
client_options
  .always_verify_peer(true)
  .openssl_ciphers(
      "ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:"
      "DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5")
  .openssl_options(SSL_OP_NO_SSLv3 | SSL_OP_NO_SSLv2 | SSL_OP_ALL);
```

Note: *"As of OpenSSL 1.0.1s the SSL_OP_NO_SSLv2 option is set by default."*

I'd also recommend SNI and certificate verification (depending on your use-case).

You may also want to consider testing against https://www.ssllabs.com/ssltest/viewMyClient.html.

**Comment by idotta** [comment]
@anonimal  thanks a lot! I'll try to follow these instructions.
